components/eam/src/chemistry/mozart/mo_drydep.F90

```diff
@@ -2523,723 +520,684 @@ subroutine drydep_xactive( ncdate, sfc_temp, pressure_sfc,  &
-                      rsmx(i,lt,ispec) = (dewm*rs*drat(m) + rmx)
-                      !-------------------------------------------------------------------------------------
-                      ! jfl : special case for PAN
-                      !-------------------------------------------------------------------------------------
-                      if( ispec == pan_ndx .or. ispec == xpan_ndx ) then
-                         dv_pan =  c0_pan(lt) * (1._r8 - exp( -k_pan(lt)*(dewm*rs*drat(m))*1.e-2_r8 ))
-                         if( dv_pan > 0._r8 .and. sndx /= 4 ) then
-                            rsmx(i,lt,ispec) = ( 1._r8/dv_pan )
-                         end if
-                      end if
-                      rclx(i,lt,ispec) = cts(i) + 1._r8/((heff(i,m)/(1.e5_r8*rcls(sndx,lt))) + (foxd(m)/rclo(sndx,lt)))
-                      rlux(i,lt,ispec) = cts(i) + rlu(sndx,lt)/(1.e-5_r8*heff(i,m) + foxd(m))
-                   end if
-                end if
-             end do
-          end do
-       end if
-    end do species_loop1
+                      endif
+                      rsmx(icol,lt,ispec) = (dewm*rs*drat(idx_drydep) + rmx)
+                   endif
+                endif
+             enddo
+          enddo
+       endif
+    enddo
+  end subroutine calculate_resistance_rgsx_and_rsmx
+
+
+  subroutine calculate_resistance_rclx(ncol, beglt, endlt, index_season, fr_lnduse, heff, cts, &   ! in
+                                       rclx)                                                       ! out
+
+    use seq_drydep_mod, only: rclo, rcls, rlu, foxd
+
+    ! input
+    integer, intent(in) :: ncol
+    integer, intent(in) :: beglt
+    integer, intent(in) :: endlt
+    integer, intent(in) :: index_season(ncol, n_land_type)
+    logical, intent(in) :: fr_lnduse(ncol, n_land_type)
+    real(r8), intent(in) :: heff(ncol,nddvels)                ! Henry's law coefficients
+    real(r8), intent(in) :: cts(ncol)                         ! correction to rlu rcl and rgs for frost
+
+    ! output
+    real(r8), intent(out) :: rclx(ncol,n_land_type,gas_pcnst)   ! lower canopy resistance [s/m]
+
+    ! local variables
+    integer :: icol, lt, ispec, idx_drydep, sndx
+
+    
+    do ispec = 1,gas_pcnst
+       if( has_dvel(ispec) ) then
+          idx_drydep = map_dvel(ispec)
+          do lt = beglt,endlt
+             do icol = 1,ncol
+                if( fr_lnduse(icol,lt) ) then
+                   sndx = index_season(icol,lt)
+
+                   if( lt == lt_for_water ) then
+                      rclx(icol,lt,ispec) = large_value
+                   else
+                      rclx(icol,lt,ispec) = cts(icol) + 1._r8/((heff(icol,idx_drydep)/(1.e5_r8*rcls(sndx,lt))) + (foxd(idx_drydep)/rclo(sndx,lt)))
+                   endif
+                endif
+             enddo
+          enddo
+       endif
+    enddo
+
+    do ispec = 1,gas_pcnst
+       if ( has_dvel(ispec) ) then
+          if( ispec == so2_ndx ) then
```

Comment:
can you merge these two if conditions into one with .and.?

Reply:
merged